### PR TITLE
no-loc when xref-not-found

### PR DESCRIPTION
**Pull request by AngryBerryMS**
[AB#532861](https://ceapex.visualstudio.com/Engineering/_workitems/edit/532861)

### CODE REVIEW

File: src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs [hunk 1]
```diff
@@ -153,6 +153,7 @@ public static MarkdownPipelineBuilder UseTripleColon(this MarkdownPipelineBuilde
     public static MarkdownPipelineBuilder UseNoloc(this MarkdownPipelineBuilder pipeline)
     {
         pipeline.Extensions.AddIfNotAlready(new NolocExtension());
+        pipeline.Extensions.AddIfNotAlready(new NoLocXrefContainerExtension());
         return pipeline;
```
Comment (herohua) [inline src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs:157]: Would be better to put under `UseXref`?

**Review by herohua — APPROVED** [review]
:shipit:

File: src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs]: Done.

File: src/docfx/lib/markdown/XrefExtension.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ internal static class XrefExtension
 
                 if (xrefLink.Href is null)
                 {
-                    return new LiteralInline(raw);
+                    return new NoLocXrefContainer(new LiteralInline(raw));
```
Comment (yufeih) [inline src/docfx/lib/markdown/XrefExtension.cs:30]: We could use `HtmlInline` to replace this `NoLocXrefContainer`.

Comment (AngryBerryMS) [reply src/docfx/lib/markdown/XrefExtension.cs:30]: `htmlTag`="<span class=no-loc>...</span>"?

File: src/docfx/lib/markdown/XrefExtension.cs [hunk 3]
```diff
@@ -27,7 +27,9 @@ internal static class XrefExtension
 
                 if (xrefLink.Href is null)
                 {
-                    return new LiteralInline(raw);
+                    var nolocInline = new NolocInline();
```
Comment (yufeih) [inline src/docfx/lib/markdown/XrefExtension.cs:30]: inline initialize.
```suggestion
                    return new NolocInline { Text = raw };
```

**Review by yufeih — APPROVED** [review]

File: src/docfx/lib/markdown/XrefExtension.cs (hunk 2 above)
Comment (AngryBerryMS) [reply src/docfx/lib/markdown/XrefExtension.cs]: Use nolocinline now

File: src/docfx/lib/markdown/XrefExtension.cs (hunk 3 above)
Comment (AngryBerryMS) [reply src/docfx/lib/markdown/XrefExtension.cs]: Donw

**Pull request merged**